dags/hindawi/utils.py

```diff
@@ -0,0 +1,46 @@
+import uuid
+from datetime import date, datetime
+from io import BytesIO
+from xml.etree import ElementTree
+
+from airflow.api.common import trigger_dag
+
+
+def save_file_in_s3(data, repo):
+    if bool(data):
+        date_today = date.today().strftime("%Y-%m-%d")
+        prefix = datetime.now().strftime("%Y-%m-%dT%H:%M")
+        key = f"{date_today}/{prefix}.json"
+        byte_file = BytesIO(data)
+        repo.save(key, byte_file)
+        return key
+
+
+def split_xmls(repo, key):
+    ids_and_records = []
+    file = repo.find_by_id(key)
+    xml_string = file.getvalue().decode("utf-8")
+    for tag in ElementTree.fromstring(xml_string):
+        if "ListRecords" in tag.tag:
```

Comment:
```
if "ListRecords" in tag.tag:
    continue
```

Reply:
This one I didn't really get it 😅 could you explain it a little bit more?

Reply:
sorry, I meant:
```
if "ListRecords" not in tag.tag:
    continue
```
to remove one indentation level. Then you can continue the rest of the code in the loop without an indentation, so
```
    for tag in ElementTree.fromstring(xml_string):
        if "ListRecords" in tag.tag:
             continue
        for record in tag:
             ....
 ```

Reply:
I don't think that's the case. The idea of this is that the _record_ is in the ListRecords tag, and the only one way to check the name of the tag is by tag.tag. The data looks like this:
```xml
<ListRecords>
   <rocord>...</record>
   <rocord>...</record>
</ListRecords >
<somethingElse>....<somethingElse/>
```
So if I will take your suggestion, I will grab _somethingElse_ instead of _record_. And a _record_ I need

Reply:
yes, but you run your code when tag.tag is in `ListRecords`. You just continue to next tag if `ListRecords` in the `tag.tag`. It doesn't change anything in the code logic.

Reply:
aaa, I see now! **not in** , I skip the word **not**. Thank you for clarifying!